tools/c7n_tencentcloud/c7n_tencentcloud/resources/cos.py

```diff
@@ -0,0 +1,355 @@
+        return results
+
+    def process_bucket(self, b):
+        rules = []
+        try:
+            resp = self.manager.source.get_cos_client(b["Location"]).get_bucket_encryption(
+                Bucket=b['Name'])
+            rules = resp["Rule"]
+        except CosServiceError as e:
+            if not e.get_error_code().startswith("NoSuch"):
+                self.log.error('error cos client error:%s\n%s', b['Name'], e.get_error_msg())
+
+        if self.data.get('state', True):
+            for sse in rules:
+                return self.filter_bucket(sse)
+            return False
+        else:
+            # If there is no configuration information,rules==[]
+            return rules is []
+
+    def filter_bucket(self, sse):
+        # only support ASE256 now
+        crypto = self.data.get('crypto')
+        if not crypto:
+            return True
+
+        rule = sse['ApplyServerSideEncryptionByDefault']
+        return crypto == rule['SSEAlgorithm']
+
+
+@COS.filter_registry.register('bucket-logging')
+class BucketLoggingFilter(BucketFilterBase):
+    """
+    Filter based on bucket logging configuration
+
+    :example:
+
+    .. code-block:: yaml
+        policies:
+        - name: bucket-logging
+          resource: tencentcloud.cos
+          filters:
+            - type: bucket-logging
+              op: enabled
+
+    """
+    schema = type_schema(
+        'bucket-logging',
+        op={'enum': ['enabled', 'disabled', 'equal', 'not-equal', 'eq', 'ne']},
+        required=['op'],
+        target_bucket={'type': 'string'},
+        target_prefix={'type': 'string'})
+    schema_alias = False
+
+    def get_logging(self, b):
+        try:
+            return self.manager.source.get_cos_client(b["Location"]).get_bucket_logging(
+                Bucket=b['Name'])
+        except CosServiceError as e:
+            if not e.get_error_code().startswith("NoSuch"):
+                self.log.error('error cos client error:%s\n%s', b['Name'], e.get_error_msg())
+            return {}
+
+    def process(self, buckets, event=None):
+        return list(filter(None, map(self.process_bucket, buckets)))
+
+    def process_bucket(self, b):
+        if self.match_bucket(b):
+            return b
+
+    def match_bucket(self, b):
+        op = self.data.get('op')
+        config_logging = self.get_logging(b)
+
+        if op == 'disabled':
+            return config_logging == {}
+        elif op == 'enabled':
+            return config_logging != {}
+        config_logging = config_logging.get("LoggingEnabled")
+
+        variables = self.get_std_format_args(b)
+        variables.update({
+            'account': self.manager.config.account_id,
+            'source_bucket_name': b['Name'],
+            'source_bucket_region': b['Location'],
+            'target_bucket_name': self.data.get('target_bucket'),
+            'target_prefix': self.data.get('target_prefix'),
+        })
+        data = format_string_values(self.data, **variables)
```

Comment:
should copy self.data to ensure we dont overwrite the existing `self.data`

Reply:
has been modified